core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -63,7 +63,7 @@ private[log] object ProducerStateEntry {
   private[log] val NumBatchesToRetain = 5
 
   def empty(producerId: Long) = new ProducerStateEntry(producerId,
-    batchMetadata = mutable.Queue[BatchMetadata](),
+    batchMetadata = new mutable.Queue[BatchMetadata](5),
```

Comment:
minor: you may want to have it as `new mutable.Queue[BatchMetadata](NumBatchesToRetain)`  instead of harcoding directly.

Reply:
Yes, good suggestion :)